### PR TITLE
Move alternatives out of String.Split ref

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to dotnet/dotnet-api-docs#4309.

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/standard/base-types/parse-strings?branch=pr-en-us-21288).

### CODE REVIEW

**Comment by gewarren** [comment]
Close/reopen.

File: docs/standard/base-types/parse-strings.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+```text
+[This is captured\ntext.]\n\n[\n[This is more captured text.]\n]
+\n[Some more captured text:\n   Option1\n   Option2][Terse text.]
+```
+
+A regular expression can split this string easily, as the following example shows.
+
+:::code language="csharp" source="snippets/parse-strings/csharp/regex.cs" id="2" interactive="try-dotnet":::
+:::code language="visual-basic" source="snippets/parse-strings/vb/regex.vb" id="2":::
```
Comment (adegeo) [inline docs/standard/base-types/parse-strings.md:67]:
```suggestion
:::code language="vb" source="snippets/parse-strings/vb/regex.vb" id="2":::
```

File: docs/standard/base-types/parse-strings.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+<xref:System.StringSplitOptions.RemoveEmptyEntries?displayProperty=nameWithType> for the `options` parameter.
+
+[!code-csharp-interactive[Intro#1](snippets/parse-strings/csharp/intro.cs#3)]
+
+## Regular expressions
+
+If your string conforms to a fixed pattern, you can use a regular expression to extract and handle its elements. For example, if strings take the form "*number* *operand* *number*", you can use a [regular expression](/dotnet/standard/base-types/regular-expressions) to extract and handle the string's elements. Here's an example:
+
+:::code language="csharp" source="snippets/parse-strings/csharp/regex.cs" id="1" interactive="try-dotnet":::
+:::code language="visual-basic" source="snippets/parse-strings/vb/regex.vb" id="1":::
```
Comment (adegeo) [inline docs/standard/base-types/parse-strings.md:39]:
```suggestion
:::code language="vb" source="snippets/parse-strings/vb/regex.vb" id="1":::
```

File: docs/standard/base-types/parse-strings.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+|Pattern|Description|
+|-------------|-----------------|
+|`\[`|Match an opening bracket.|
+|`([^\[\]]+)`|Match any character that is not an opening or a closing bracket one or more times. This is the first capturing group.|
+|`\]`|Match a closing bracket.|
+
+The <xref:System.Text.RegularExpressions.Regex.Split%2A?displayProperty=nameWithType> method is almost identical to <xref:System.String.Split%2A?displayProperty=nameWithType>, except that it splits a string based on a regular expression pattern instead of a fixed character set. For example, the following example uses the <xref:System.Text.RegularExpressions.Regex.Split%2A?displayProperty=nameWithType> method to split a string that contains substrings delimited by various combinations of hyphens and other characters.
+
+:::code language="csharp" source="snippets/parse-strings/csharp/regex.cs" id="3" interactive="try-dotnet":::
+:::code language="visual-basic" source="snippets/parse-strings/vb/regex.vb" id="3":::
```
Comment (adegeo) [inline docs/standard/base-types/parse-strings.md:80]:
```suggestion
:::code language="vb" source="snippets/parse-strings/vb/regex.vb" id="3":::
```

File: docs/standard/base-types/parse-strings.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+- <xref:System.String.IndexOfAny%2A>, which returns the zero-based index in the current string instance of the first occurrence of any character in a character array.
+
+- <xref:System.String.LastIndexOf%2A>, which returns the zero-based index of the last occurrence of a character or string in a string instance.
+
+- <xref:System.String.LastIndexOfAny%2A>, which returns a zero-based index in the current string instance of the last occurrence of any character in a character array.
+
+The following example uses the <xref:System.String.IndexOf%2A> method to find the periods in a string. It then uses the <xref:System.String.Substring%2A> method to return full sentences.
+
+:::code language="csharp" source="snippets/parse-strings/csharp/indexof.cs" id="1" interactive="try-dotnet":::
+:::code language="visual-basic" source="snippets/parse-strings/vb/indexof.vb" id="1":::
```
Comment (adegeo) [inline docs/standard/base-types/parse-strings.md:107]:
```suggestion
:::code language="vb" source="snippets/parse-strings/vb/indexof.vb" id="1":::
```

Comment (adegeo) [inline docs/standard/base-types/parse-strings.md:107]: This ends abruptly; perhaps **see also** with link back to overview or some other string related examples?

File: docs/standard/base-types/parse-strings.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  - "csharp"
+  - "vb"
+helpviewer_keywords:
+  - "strings [.NET], breaking up"
+---
+# Extract elements from a string
+
+This article covers some different techniques for extracting parts of a string.
+
+## String.Split method
```
Comment (adegeo) [inline docs/standard/base-types/parse-strings.md:16]: This section is missing VB versions.

**Review by adegeo — CHANGES_REQUESTED** [review]
- Some minor things to fix and a thought about the end of the article.
- Code snippets aren't showing correctly because of the wrong code language.

**Review by adegeo — APPROVED** [review]
LGTM!

**Comment by gewarren** [comment]
@adegeo Apologies for requesting review and then adding a couple more commits. It was just things I noticed on my final review. I added an intro and a couple of links.

**Pull request merged**